<?xml version="1.0"?>

<launch>
  <param name="/use_sim_time" value="false" />

  <arg name="global_frame" value="map" />

  <arg name="map_file" default="$(find lattice_path_planner)/map/map.yaml" />

  <node pkg="lattice_path_planner" type="lattice_path_planner_test" name="lattice_path_planner_test" output="screen">
    <!-- global_costmap -->
    <rosparam file="$(find lattice_path_planner)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find lattice_path_planner)/params/global_costmap_params.yaml" command="load" />
    <param name="global_costmap/global_frame" value="$(arg global_frame)" />

    <!-- LatticePathPlannerROS -->
    <rosparam file="$(find lattice_path_planner)/params/lattice_path_planner_params.yaml" command="load" />
    <param name="LatticePathPlannerROS/primitive_filename" value="$(find lattice_path_planner)/params/pr2_5cm.mprim" />
  </node>

  <!-- tf -->
  <node pkg="lattice_path_planner" type="lattice_path_planner_tf_broadcaster" name="lattice_path_planner_tf_broadcaster" output="screen"/>

  <!-- Mapping -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id" value="$(arg global_frame)" />
  </node>

  <!-- Visualization -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find lattice_path_planner)/launch/demo.rviz" />

</launch>
